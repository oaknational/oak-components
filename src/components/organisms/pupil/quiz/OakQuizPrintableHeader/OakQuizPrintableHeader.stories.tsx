import React from "react";
import { StoryObj, Meta } from "@storybook/react";

import { OakQuizPrintableHeader } from "./OakQuizPrintableHeader";

const meta: Meta<typeof OakQuizPrintableHeader> = {
  component: OakQuizPrintableHeader,
  tags: ["autodocs"],
  argTypes: {
    iconName: {},
  },
  parameters: {
    controls: {
      include: ["iconBackground", "iconName", "alt", "breadcrumbs", "title"],
    },
  },
  args: {
    iconName: "subject-science",
    alt: "icon",
    breadcrumbs: ["first", "second", "third", "fourth"],
    worksheetDownloaded: true,
    videoPercentage: 80,
  },
};

export default meta;

type Story = StoryObj<typeof OakQuizPrintableHeader>;

export const Default: Story = {
  render: (args) => <OakQuizPrintableHeader {...args} />,
  args: {},
};
