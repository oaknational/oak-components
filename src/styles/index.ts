export * from "@/styles/theme/color";
