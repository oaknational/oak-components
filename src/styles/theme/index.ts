export { oakColorTokens, oakUiRoleTokens } from "./color";

export type {
  OakColorToken,
  OakCombinedColorToken,
  OakUiRoleToken,
  OakColorFilterToken,
} from "./color";

export {
  oakAllSpacingTokens,
  oakInnerPaddingTokens,
  oakSpaceBetweenTokens,
} from "./spacing";

export type {
  OakAllSpacingToken,
  OakCombinedSpacingToken,
  OakInnerPaddingToken,
  OakSpaceBetweenToken,
} from "./spacing";

export { oakDropShadowTokens } from "./dropShadow";
export type { OakDropShadowToken } from "./dropShadow";

export { oakOpacityTokens } from "./opacity";
export type { OakOpacityToken } from "./opacity";

export { oakTransitionTokens } from "./transitions";
export type { OakTransitionToken } from "./transitions";

export { oakBorderRadiusTokens, oakBorderWidthTokens } from "./borders";
export type { OakBorderRadiusToken, OakBorderWidthToken } from "./borders";

export type { OakTheme } from "./theme";

export { oakFontTokens, oakFontSizeTokens } from "./typography";
export type { OakFontToken, OakFontSizeToken } from "./typography";

export { oakZIndexTokens } from "./zIndex";

export { oakDefaultTheme } from "./default.theme";
