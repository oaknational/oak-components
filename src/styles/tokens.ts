export const oakAllSpacingPx = {
  "all-spacing-1": 4,
  "all-spacing-2": 8,
  "all-spacing-3": 12,
  "all-spacing-4": 16,
  "all-spacing-5": 20,
  "all-spacing-6": 24,
  "all-spacing-7": 32,
  "all-spacing-8": 40,
  "all-spacing-9": 48,
  "all-spacing-10": 56,
  "all-spacing-11": 64,
  "all-spacing-12": 72,
  "all-spacing-13": 80,
  "all-spacing-14": 92,
  "all-spacing-15": 100,
  "all-spacing-16": 120,
};

export type OakAllSpacing = keyof typeof oakAllSpacingPx;

export const oakInnerPaddingAllSpacing = {
  "inner-padding-xs": "all-spacing-2",
  "inner-padding-s": "all-spacing-3",
  "inner-padding-m": "all-spacing-4",
  "inner-padding-l": "all-spacing-5",
  "inner-padding-xl": "all-spacing-6",
};

export type OakInnerPadding = keyof typeof oakInnerPaddingAllSpacing;

export const oakSpaceBetweenAllSpacing = {
  "space-between-ssx": "all-spacing-2",
  "space-between-xs": "all-spacing-3",
  "space-between-s": "all-spacing-4",
  "space-between-m": "all-spacing-6",
  "space-between-m2": "all-spacing-7",
  "space-between-l": "all-spacing-9",
  "space-between-xl": "all-spacing-10",
  "space-between-xxl": "all-spacing-12",
  "space-between-xxxl": "all-spacing-13",
};

export type OakSpaceBetween = keyof typeof oakSpaceBetweenAllSpacing;

export const oakAllColorsHex = {
  white: "#ffffff",
  grey20: "#f2f2f2",
  grey30: "#e4e4e4",
  grey40: "#cacaca",
  grey50: "#808080",
  grey60: "#575757",
  grey70: "#2d2d2d",
  black: "#222222",
  oakGreen: "#287c34",
  mint: "bef2bd",
  mint30: "ebfbeb",
  mint50: "dff9de",
  aqua: "#b0e2de",
  aqua30: "#e7f6f5",
  aqua50: "#cee7e5",
  lavender: "#a0b6f2",
  lavender30: "#e3e9fb",
  lavender50: "#c6d1ef",
  pink: "#deb7d5",
  pink30: "#f5e9f2",
  pink50: "#e5d1e0",
  lemon: "#ffe555",
  lemon30: "#fff7cc",
  lemon50: "#f6e8a0",
  amber: "#ff934e",
  amber30: "#ffdfca",
  amber50: "#ffc8a6",
  red: "#dd0035",
  navy: "#0d24c4",
  blue: "#374cf1",
  magenta: "#d02aa7",
  purple: "#845ad9",
  teal: "#037b7d",
};

export type OakColor = keyof typeof oakAllColorsHex;

export const oakColorTokens = [
  "text-primary",
  "text-subdued",
  "text-error",
  "text-disabled",
  "text-link-active",
  "text-link-hover",
  "text-link-visited",
  "text-inverted",
  "text-success",
  "text-warning",
  "bg-primary",
  "bg-neutral",
  "bg-neutral-50",
  "bg-neutral-30",
  "bg-btn-primary",
  "bg-btn-primary-hover",
  "bg-btn-primary-disabled",
  "bg-btn-secondary",
  "bg-btn-secondary-hover",
  "bg-btn-secondary-disabled",
  "bg-icon",
  "bg-icon-hover",
  "bg-decorative1â€“main",
  "bg-decorative1-subdued",
  "bg-decorative1-very subdued",
  "bg-decorative2-main",
  "bg-decorative2-subdued",
  "bg-decorative2-very-subdued",
  "bg-decorative3",
  "bg-decorative3-50",
  "bg-decorative3-30",
  "bg-decorative4",
  "bg-decorative4-50",
  "bg-decorative4-30",
  "bg-decorative5",
  "bg-decorative5-50",
  "bg-decorative5-30",
  "bg-decorative6",
  "bg-decorative6-50",
  "bg-decorative6-30",
  "border-primary",
  "border-inverted",
  "border-neutral",
  "border-decorative1",
  "border-decorative2",
  "border-decorative3",
  "border-decorative4",
  "border-decorative5",
  "border-decorative6",
] as const;

export type OakColorToken = (typeof oakColorTokens)[number];
