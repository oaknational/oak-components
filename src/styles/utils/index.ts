export * from "./colorStyle";
export * from "./positionStyle";
export * from "./sizeStyle";
export * from "./spacingStyle";
export * from "./flexStyle";
