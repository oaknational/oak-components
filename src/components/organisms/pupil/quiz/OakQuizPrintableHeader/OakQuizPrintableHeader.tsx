import React from "react";

import {
  OakBulletList,
  OakBulletListProps,
  OakHandDrawnCardWithIcon,
  OakJauntyAngleLabel,
} from "@/components/molecules";
import { OakFlex, OakHeading, OakIconProps } from "@/components/atoms";
import { FlexStyleProps } from "@/styles/utils/flexStyle";

export type OakQuizPrintableHeaderProps = {
  title: string;
  iconName: OakIconProps["iconName"];
  alt?: OakIconProps["alt"];
  breadcrumbs: OakBulletListProps["listItems"];
  optionalityTitle?: string;
  worksheetDownloaded: boolean;
  workSheetAvailable: boolean;
  videoPercentage: number;
} & FlexStyleProps;

/**
 * This component is the header for the printable view;
 *
 *
 *
 */
export const OakQuizPrintableHeader = ({
  title,
  iconName,
  alt,
  breadcrumbs,
  worksheetDownloaded,
  workSheetAvailable,
  videoPercentage,
}: OakQuizPrintableHeaderProps) => {
  return (
    <OakFlex
      $flexDirection={["column", "row"]}
      $gap={"space-between-s"}
      $justifyContent={"space-between"}
    >
      <OakFlex $flexDirection={"row"} $gap={"space-between-s"}>
        <OakHandDrawnCardWithIcon
          iconName={iconName}
          fill={"white"}
          $flexGrow={0}
          iconHeight={"all-spacing-10"}
          iconWidth={"all-spacing-10"}
          $width={"all-spacing-10"}
          $height={"all-spacing-10"}
          alt={alt}
        />
        <OakFlex $flexDirection={"column"} $gap={"space-between-xs"}>
          <OakHeading tag="h1" $font={"heading-7"}>
            {title}
          </OakHeading>
          <OakBulletList listItems={breadcrumbs} />
        </OakFlex>
      </OakFlex>
      <OakFlex
        $flexDirection={"column"}
        $gap={"space-between-xs"}
        $alignItems={"flex-start"}
      >
        <OakJauntyAngleLabel
          label={`Lesson video completion - ${videoPercentage}%`}
          $background={"bg-neutral"}
          $font={"heading-light-7"}
        />
        <OakJauntyAngleLabel
          label={
            workSheetAvailable
              ? `Worksheet downloaded - ${worksheetDownloaded ? "Yes" : "No"}`
              : "No worksheet"
          }
          $background={"bg-neutral"}
          $font={"heading-light-7"}
        />
      </OakFlex>
    </OakFlex>
  );
};
