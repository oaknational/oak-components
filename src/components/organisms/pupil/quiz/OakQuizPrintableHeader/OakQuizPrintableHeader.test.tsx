import React from "react";
import "@testing-library/jest-dom";
import { create } from "react-test-renderer";
import { ThemeProvider } from "styled-components";

import { OakQuizPrintableHeader } from "./OakQuizPrintableHeader";

import renderWithTheme from "@/test-helpers/renderWithTheme";
import { oakDefaultTheme } from "@/styles";

describe("OakQuizPrintableHeader", () => {
  it("renders", () => {
    const { getByText } = renderWithTheme(
      <ThemeProvider theme={oakDefaultTheme}>
        <OakQuizPrintableHeader
          iconName="subject-science"
          alt="icon"
          breadcrumbs={["first", "second", "third", "fourth"]}
          title="Pupil Journey Header"
          worksheetDownloaded={true}
          videoPercentage={80}
          data-testid="test"
          workSheetAvailable={true}
        />
      </ThemeProvider>,
    );
    expect(getByText("Pupil Journey Header")).toBeInTheDocument();
  });
  it("renders worksheet downloaded when applicable", () => {
    const { getByText } = renderWithTheme(
      <ThemeProvider theme={oakDefaultTheme}>
        <OakQuizPrintableHeader
          iconName="subject-science"
          alt="icon"
          breadcrumbs={["first", "second", "third", "fourth"]}
          title="Pupil Journey Header"
          worksheetDownloaded={true}
          videoPercentage={80}
          data-testid="test"
          workSheetAvailable={true}
        />
      </ThemeProvider>,
    );
    expect(getByText("Worksheet downloaded - Yes")).toBeInTheDocument();
  });
  it("renders worksheet not downloaded when applicable", () => {
    const { getByText } = renderWithTheme(
      <ThemeProvider theme={oakDefaultTheme}>
        <OakQuizPrintableHeader
          iconName="subject-science"
          alt="icon"
          breadcrumbs={["first", "second", "third", "fourth"]}
          title="Pupil Journey Header"
          worksheetDownloaded={false}
          videoPercentage={80}
          data-testid="test"
          workSheetAvailable={true}
        />
      </ThemeProvider>,
    );
    expect(getByText("Worksheet downloaded - No")).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const tree = create(
      <ThemeProvider theme={oakDefaultTheme}>
        {" "}
        <OakQuizPrintableHeader
          iconName="subject-science"
          alt="icon"
          breadcrumbs={["first", "second", "third", "fourth"]}
          title="Pupil Journey Header"
          worksheetDownloaded={true}
          videoPercentage={80}
          workSheetAvailable={true}
        />
      </ThemeProvider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
