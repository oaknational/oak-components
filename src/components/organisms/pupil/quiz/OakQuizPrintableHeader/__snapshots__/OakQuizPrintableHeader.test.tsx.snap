// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OakQuizPrintableHeader matches snapshot 1`] = `
[
  " ",
  .c0 {
  font-family: --var(google-font),Lexend,sans-serif;
}

.c3 {
  position: relative;
  width: 3.5rem;
  height: 3.5rem;
  padding: 1.5rem;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c5 {
  position: absolute;
  inset: 0rem;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c8 {
  position: relative;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c9 {
  position: relative;
  width: 3.5rem;
  min-width: 3.5rem;
  height: 3.5rem;
  min-height: 3.5rem;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c14 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
  background: #f2f2f2;
  border-radius: 0.375rem;
  -webkit-transform: rotate(-1.5deg);
  -ms-transform: rotate(-1.5deg);
  transform: rotate(-1.5deg);
  font-family: --var(google-font),Lexend,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.25rem;
  -webkit-letter-spacing: 0.0115rem;
  -moz-letter-spacing: 0.0115rem;
  -ms-letter-spacing: 0.0115rem;
  letter-spacing: 0.0115rem;
}

.c10 {
  -webkit-filter: invert(10%) sepia(1%) saturate(236%) hue-rotate(314deg) brightness(95%) contrast(91%);
  filter: invert(10%) sepia(1%) saturate(236%) hue-rotate(314deg) brightness(95%) contrast(91%);
  object-fit: contain;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  gap: 1rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 1rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.75rem;
}

.c12 {
  font-family: --var(google-font),Lexend,sans-serif;
  font-weight: 600;
  font-size: 1rem;
  line-height: 1.25rem;
  -webkit-letter-spacing: 0.0115rem;
  -moz-letter-spacing: 0.0115rem;
  -ms-letter-spacing: 0.0115rem;
  letter-spacing: 0.0115rem;
}

.c13 {
  font-family: --var(google-font),Lexend,sans-serif;
}

.c6 > * {
  width: 100%;
  height: 100%;
}

.c7 {
  fill: #ffffff;
}

<div
    className="c0 c1"
  >
    <div
      className="c0 c2"
    >
      <div
        className="c3 c4"
      >
        <div
          aria-hidden={true}
          className="c5 c6"
        >
          <svg
            className="c7"
            preserveAspectRatio="none"
            viewBox="0 0 457 120"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M27.442 2.59c5.262-.362 11.96-.656 18.624-.932 4.374-.19 8.748-.371 12.712-.578 1.367-.07 3.042-.087 4.716-.095C65.032.976 66.57.959 67.903.907 91.858-.017 115.164.268 139.325.57l6.22.078c20.333.12 40.768.086 61.033-.13 6.664-.025 13.294-.043 19.957-.06C243.656.415 260.777.363 277.83.234c8.85-.07 17.667-.052 26.518-.035 12.132.026 24.366.052 36.771-.147 8.475-.138 17.633.087 26.757.302 3.725.087 7.45.182 11.107.25 3.69.061 7.347.139 10.969.217 4.511.095 8.988.19 13.499.259 13.464.207 30.619 1.519 30.619 1.519 3.588.449 7.074 1.32 10.286 2.59 2.973 1.304 4.682 4.135 6.391 6.967.717 1.165 1.435 2.33 2.221 3.384.273.363.342.812.376 1.26.034.406.102.82.307 1.175 3.52 6.388 2.427 12.604 1.299 19.044a27.243 27.243 0 0 0-.273 1.658 70.082 70.082 0 0 0 .478 16.273l.205 5.31c.171 4.575.342 9.133.854 13.683.274 2.357.205 4.705.137 7.07-.102 3.238-.205 6.492.547 9.808.547 2.261-.308 4.705-1.162 7.13-.342.993-.684 1.986-.923 2.962-.239.984-.546 1.951-.82 2.926-.342 1.2-.718 2.392-.991 3.6-1.196 5.379-4.511 8.435-11.756 9.134-2.392.233-4.75.518-7.142.855h-6.937c-5.263.371-11.961.656-18.624.932-4.375.19-8.749.371-12.713.579-1.367.069-3.041.086-4.716.095-1.538.008-3.075.025-4.408.086-23.956.924-47.262.639-71.422.337l-6.186-.078a3693.945 3693.945 0 0 0-61.033.129c-6.664.026-13.294.044-19.958.061-17.121.043-34.241.095-51.294.224-8.851.069-17.667.052-26.518.035-12.132-.026-24.366-.052-36.771.147-8.475.138-17.633-.087-26.758-.303-3.725-.086-7.45-.181-11.106-.25-3.69-.06-7.313-.138-10.97-.216-4.51-.095-8.987-.19-13.498-.259-18.932-.293-30.175-1.079-33.66-2.754-1.47-.38-2.871-.837-4.238-1.355-3.076-1.304-4.887-4.135-6.664-6.967-.752-1.165-1.47-2.331-2.29-3.384-.273-.363-.341-.812-.376-1.261-.034-.405-.102-.82-.307-1.174C0 95.664 1.128 89.448 2.324 83.008l.308-1.658c.478-5.421.341-10.869-.513-16.273-.102-1.77-.17-3.548-.24-5.318-.17-4.567-.375-9.125-.888-13.675-.273-2.357-.205-4.705-.136-7.07.102-3.238.205-6.493-.581-9.808-.547-2.262.341-4.705 1.196-7.13.342-.994.717-1.986.99-2.962.24-.984.548-1.96.855-2.927.376-1.2.752-2.39 1.025-3.6C5.605 7.21 9.022 4.154 16.574 3.455c2.495-.233 10.936-.855 10.936-.855l-.069-.009Z"
              vectorEffect="non-scaling-stroke"
            />
          </svg>
        </div>
        <div
          className="c8"
        >
          <div
            className="c9"
          >
            <img
              alt="icon"
              className="c10"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="https://res.cloudinary.com/mock-cloudinary-cloud/image/upload/subject-icons/science.svg"
              style={
                {
                  "bottom": 0,
                  "color": "transparent",
                  "height": "100%",
                  "left": 0,
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": "100%",
                }
              }
            />
          </div>
        </div>
      </div>
      <div
        className="c0 c11"
      >
        <h1
          className="c12"
        >
          Pupil Journey Header
        </h1>
        <span
          className="c13"
        >
          <span
            className="c13"
          >
            <span
              className="c13"
            >
              first
            </span>
          </span>
          <span
            className="c13"
          >
            <span
              className="c13"
            >
               • 
            </span>
            <span
              className="c13"
            >
              second
            </span>
          </span>
          <span
            className="c13"
          >
            <span
              className="c13"
            >
               • 
            </span>
            <span
              className="c13"
            >
              third
            </span>
          </span>
          <span
            className="c13"
          >
            <span
              className="c13"
            >
               • 
            </span>
            <span
              className="c13"
            >
              fourth
            </span>
          </span>
        </span>
      </div>
    </div>
    <div
      className="c0 c11"
    >
      <div
        className="c14"
      >
        Lesson video completion - 80%
      </div>
      <div
        className="c14"
      >
        Worksheet downloaded - Yes
      </div>
    </div>
  </div>,
]
`;
